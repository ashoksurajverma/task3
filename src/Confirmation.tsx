import React, { useState, useEffect } from "react";
import "./confirmation.css";

const alertClasses = {
  primary: "alert alert-primary",
  secondary: "alert alert-secondary",
  success: "alert alert-success",
  error: "alert alert-danger",
  caution: "alert alert-warning",
  message: "alert alert-info",
  light: "alert alert-light",
  dark: "alert alert-dark",
};
interface IProps {
  message: string;
  type:
    | "primary"
    | "secondary"
    | "success"
    | "error"
    | "caution"
    | "message"
    | "light"
    | "dark";
  acceptLabel: string;
  declineLabel: string;
  accept: () => void;
  decline: () => void;
}

function Confirmation({
  message,
  type,
  accept,
  decline,
  acceptLabel,
  declineLabel,
}: IProps) {
  const [shouldRender, setShouldRender] = useState(true);

  const onClickHandle = (isCofirmation: boolean) => {
    if (isCofirmation) {
      accept();
    } else {
      decline();
    }
    setShouldRender(false);
  };

  const getClassName = () => {
    if (!type) return alertClasses.message;
    return alertClasses[type];
  };

  if (!shouldRender) return null;

  return (
    <div className={getClassName()}>
      <p>{message}</p>
      <div className="btn-container">
        <div onClick={() => onClickHandle(true)} className="btn btn-primary">
          {acceptLabel}
        </div>
        <div onClick={() => onClickHandle(false)} className="btn btn-danger">
          {declineLabel}
        </div>
      </div>
    </div>
  );
}

export default Confirmation;
