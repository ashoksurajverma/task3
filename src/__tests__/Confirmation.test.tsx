import React from "react";
import { ShallowWrapper, shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Confirmation from "../Confirmation";

configure({ adapter: new Adapter() });

describe("Render Confirmation component", () => {
  let wrapper: ShallowWrapper<undefined, undefined>;

  beforeEach(() => {
    wrapper = shallow(
      <Confirmation
        accept={() => {}}
        decline={() => {}}
        message="Reveal the answer?"
        type="message"
        acceptLabel="Yes Please"
        declineLabel="Not Yet"
      />
    );
  });

  it("should render without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render btn container", () => {
    const btnContainer = wrapper.find(".btn-container").length;
    expect(btnContainer).toBe(1);
    expect(wrapper.prop("className")).toEqual("alert alert-info");
  });

  it("click on confirm btn", () => {
    const confirmBtn = wrapper.find(".btn-primary");
    confirmBtn.simulate("click");
    const btnContainer = wrapper.find(".btn-container").length;
    expect(btnContainer).toBe(0);
  });
});
