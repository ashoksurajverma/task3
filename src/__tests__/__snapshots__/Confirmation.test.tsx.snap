// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Confirmation component should render without error 1`] = `ShallowWrapper {}`;
